content/en/docs/tasks/job/job-with-pod-to-pod-communication.md

```diff
@@ -0,0 +1,106 @@
+
+Pods within a Job might need to communicate among themselves. They could query the Kubernetes API
+to learn the IPs of the other Pods, but it's much simpler to rely on Kubernetes' built-in DNS resolution.
+Jobs in Indexed completion mode automatically set the pods' hostname to be in the format of
+`${jobName}-${completionIndex}`, which can be used to deterministically determine
+pod hostnames and enable pod communication *without* needing to create a client connection to
+the Kubernetes control plane to obtain pod hostnames/IPs via API requests. This can be useful
+for use cases where pod networking is required but we don't want to depend on a network 
+connection with the Kubernetes API server.
+
+## {{% heading "prerequisites" %}}
+
+You should already be familiar with the basic use of [Job](/docs/concepts/workloads/controllers/job/).
+
+{{< include "task-tutorial-prereqs.md" >}} {{< version-check >}}
+
+{{<note>}} If you are using MiniKube or a similar tool, you may need to take [extra steps](https://minikube.sigs.k8s.io/docs/handbook/addons/ingress-dns/) to ensure you have DNS. {{</note>}}
+
+<!-- steps -->
+
+## Starting a Job with Pod-to-Pod Communication
+
+To enable pod-to-pod communication using pod hostnames in a Job, you must do the following:
+
+1. Set up a [headless service](https://kubernetes.io/docs/concepts/services-networking/service/#headless-services)
+with a valid label selector for the pods created by your Job. This will trigger the 
+Kubernetes DNS server to cache the hostnames of 
+the pods running your Job (note that the service must be in the same namespace as 
+the Job, however, the DNS server does not need to be). One easy way to do this is to use the `job-name: <your-job-name>` selector, since the `job-name` label will be automatically added by Kubernetes. 
+
+2. Include the following your Job template spec: `subdomain: <headless-svc-name>`
+   where `<headless-svc-name>` must match the name of your headless service
+   exactly. 
+
+Example Job which completes only after all pods successfully ping each other using hostnames:
+
+```yaml
+
+apiVersion: v1
+kind: Service
+metadata:
+  name: headless-svc
+spec:
+  clusterIP: None # clusterIP must be None to create a headless service
+  selector:
+    job-name: example-job # must match Job name
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: example-job
+spec:
+  completions: 3
+  parallelism: 3
+  completionMode: Indexed
+  template:
+    spec:
+      subdomain: headless-svc # has to match Service name
+      restartPolicy: Never
+      containers:
+      - name: example-workload
+        image: bash:latest
+        command:
+        - bash
+        - -c
+        - |
+          for i in 0 1 2
+          do
+            gotStatus="-1"
+            wantStatus="0"             
+            while [ $gotStatus -ne $wantStatus ]
+            do                                       
+              ping -c 1 example-job-${i}.headless-svc > /dev/null 2>&1
```

Comment:
can be prefixed by the namespace as well if the pod needs to be reached from outside the namespace.